<!Doctype HTML>
<html lang="en">
     <head>
          <meta charset="UTF-8">
          <title>liquid layout</title>
          
     </head>

<body style="width: 90%">

<a href="B.Picking_OTUs_and_Denoising.html">B. Picking OTUs and Denoising</a><i><br></i>
<h2>C. Taxonomy and the OTU Table</h2>

<b><span style="color:rgb(19,79,92)">Goals</span></b><br style="color:rgb(19,79,92)"><ul style="color:rgb(19,79,92)"><li style="list-style-position:outside;list-style-type:square">Assign taxonomy to the representative sequences</li><li style="list-style-position:outside;list-style-type:square">Build an OTU table and look at it in a spreadsheet</li><li style="list-style-position:outside;list-style-type:square">Summarize taxonomy data from an OTU table</li><li style="list-style-position:outside;list-style-type:square">Plot the taxonomy summaries</li></ul><h4><a name="TOC-Assigning-Taxonomy"></a>Assigning Taxonomy</h4>Taxonomy
 is assigned to high-throughput 16S rRNA gene sequences using some kind 
of comparison to a reference database. The most basic approach might be 
doing a BLAST search and taking the top hit. However, this doesn't 
account for redundancy or give you any idea of your confidence or 
specificity. The default algorithm in QIIME is the RDP Classifier. This 
is a Bayesian classifier that incorporates information about different 
places in the taxonomic tree where the sequence might fit in, and it 
calculates the highest probability taxonomy that can be assigned with 
some specified level of confidence. This still uses a reference 
database, in this case called a <i>training set</i>. In the version of Qiime you are using, the 
default training set is the Greengenes (GG) reference database clustered
 at 97% identity, other versions of GG are available for download from <a href="http://greengenes.lbl.gov/Download/OTUs/">here</a>.<br><br>We're going to use the QIIME script <a href="http://qiime.org/scripts/assign_taxonomy.html">assign_taxonomy</a>.py as follows (may take a minute):<br><br><b><span style="color:rgb(56,118,29);font-family:courier new,monospace">assign_taxonomy.py -i rep_set.fna -o taxonomy_results/</span></b><br><br>Some of the options:<br><span>&nbsp;&nbsp;&nbsp; </span>-i [file]&nbsp; Name of the fasta file to classify (usually OTU representative seqs)<br><span>&nbsp;&nbsp;&nbsp; </span>-o [name]&nbsp; Name of directory to create for output <br><br>To
 use a custom training set like the download from Greengenes, you have 
to reference it using the -t and -r options as described <a href="http://qiime.org/scripts/assign_taxonomy.html">in the documentation</a>.<br><br>This created a new directory called taxonomy_results. Look in there with <i>ls</i>
 and you will see that it contains two files: a log file, and a txt file
 of results. The rep_set_tax_assignments.txt file contains an entry for 
each representative sequence, listing taxonomy to the greatest depth 
allowed by the confidence threshold (80% by default, can be changed with
 the -c option), and a column of confidence values for the deepest level
 of taxonomy shown.&nbsp; These data will be really useful, once we have
 them inside an OTU table!<br><h4><a name="TOC-Building-an-OTU-Table"></a>Building an OTU Table<br></h4>An
 OTU table is a form of your sequencing results that will finally be 
really useful to analyze in excel, visualize, etc.&nbsp; It is a table 
giving the count of the number of sequences in each OTU, for each 
sample, and the taxonomy of that OTU.&nbsp; Super!<br><br>We generate an OTU table with a script called <b><a href="http://qiime.org/scripts/make_otu_table.html">make_otu_table.py</a></b> as follows<br><br><b><span style="color:rgb(56,118,29);font-family:courier new,monospace">make_otu_table.py
 -i uclust_picked_otus/inflated_denoised_seqs_otus.txt -t 
taxonomy_results/rep_set_tax_assignments.txt -o otu_table.biom</span></b><br><br>Some of the options:<br><span>&nbsp;&nbsp;&nbsp; </span>-i [file]&nbsp; The OTU map output from pick_otus.py<br><span>&nbsp;&nbsp;&nbsp; </span>-t [file]&nbsp; The taxonomy file from assign_taxonomy.py<br><span>&nbsp;&nbsp;&nbsp; </span>-o [name]&nbsp; The name of the output file<br><br>This should create an OTU table file called otu_table.biom.&nbsp; Check it out in <i>less - </i>it is in an XML/JSON format called <a href="http://biom-format.org/">biom</a> (new as of QIIME 1.5.0).&nbsp; If you want to look at the OTU counts per sample in a spreadsheet, you'll first have to <a href="http://biom-format.org/documentation/biom_conversion.html">convert</a> the biom OTU table into a normal text table.&nbsp; This can be done with the biom script as follows:<br><br><b><font color="#274e13" face="courier new, monospace">biom
 convert -i otu_table.biom -o otu_table_tabseparated.txt --to-tsv 
--header-key taxonomy --output-metadata-id "ConsensusLineage"</font></b><br><br>Now,
 the new output file otu_table_tabseparated.txt is something you can 
easily import into a spreadsheet. You can also read it more easily in 
the terminal:<br><br><b style="color:rgb(56,118,29)"><span style="font-family:courier new,monospace">less otu_table_tabseparated.txt</span></b><br><br>Note
 that this tab-separated file format for OTU tables used to be supported
 by QIIME (v 1.4 and earlier) but is no-longer a supported format as of 
QIIME 1.5.0. This is due to the needs of researchers generating 16S rRNA
 gene surveys using the Illumina HiSeq platform, with thousands of 
samples and hundreds of thousands+ of OTUs. Once the OTU table gets that
 big, it's much more reasonable to store it in the biom format.<br><br>Also
 note, when analyzing these data in a spreadsheet, that you will first 
want use the spreadsheet functions to normalize the abundances to the 
total number of reads in each sample.&nbsp; If one sample has 100 reads 
and another has 200 reads, you can't directly compare raw read numbers 
-- it's more meaningful to look at percentages.&nbsp; e.g., One OTU was 
10% of the abundance in Sample 1 <i>vs</i> 50% of the abundance in Sample 2, etc.<br><br><h4><a name="TOC-Summarizing-Taxonomy"></a>Summarizing Taxonomy</h4>The
 OTU table is pretty useful, but we can make other tables of taxonomic 
summaries that we can analyze in a spreadsheet as well.&nbsp; To 
summarize taxonomy in terms of relative abundance, we can use the script
 <b><a href="http://qiime.org/scripts/summarize_taxa.html">summarize_taxa.py</a></b> as follows:<br><br><b><span style="color:rgb(56,118,29);font-family:courier new,monospace">summarize_taxa.py -i otu_table.biom -o taxonomy_summaries/</span></b><br><br>Some of the options:<br><span>&nbsp;&nbsp;&nbsp; </span>-i [file]&nbsp; The input OTU table with included taxonomy information<br><span>&nbsp;&nbsp;&nbsp; </span>-o [name]&nbsp; The name of the directory to be created and filled with goodies<br><br>Check
 out the contents of the taxonomy_summaries folder. You can open any of 
these files in a spreadsheet and graph taxonomy summaries for your 
samples!&nbsp; The numbers are in terms of relative abundance. In other 
words, they're already normalized. The code L2, L3, etc, refers to the 
level of taxonomy summarized in the file. I broke down the L3 (class) 
taxonomy data in <a href="L3_Taxonomy_Summary.xls">this spreadsheet</a>
 to make a graph that looks like the following. The samples on the left 
are the controls, and on the right are the fasted treatments.<br>
<br>
<div style="display:block;text-align:center;margin-right:auto;margin-left:auto"><a href="L3_TaxonomyGraph2.png" imageanchor="1"><img src="L3_TaxonomyGraph2.jpeg" width="400" border="0" height="245"></a></div>
<br>By
 doing this in a spreadsheet, I can leave out all those extra 
"Other" categories (which signify unknown classification) and the small 
classes that wouldn't be visible on the graph anyway. I would just have 
to specify in the figure legend that the remaining white space 
represents unclassified reads and low-abundance classes.<br><br>If 
you're impatient and don't want to dive into the spreadsheet just yet, 
QIIME can make some simple graphs to summarize the results. The script 
that does this is called <b><a href="http://qiime.org/scripts/plot_taxa_summary.html">plot_taxa_summary.py</a></b>.&nbsp; For example, you can summarize taxonomy at the Class level (L3) like this:<br><br><b><span style="color:rgb(56,118,29);font-family:courier new,monospace">plot_taxa_summary.py -i taxonomy_summaries/otu_table_L3.txt -o taxonomy_plot_L3/</span></b><br><br>Open up the newly created bar_charts.html file in your web browser to see some graphed data. <br><br><hr width="100%" size="2"><i>Next steps</i>: <b><a href="D.Alignments_and_Trees.html">D. Alignments and Trees</a></b><br><br><br></div></div></td></tr></tbody></table>
</body></html>